### PR TITLE
FreeIPA: Fix documentation

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

* Add datatype whereever it is missing
* Fix documentation

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE 
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/fix_ipa_module_doc_type.yml
lib/ansible/modules/identity/ipa/ipa_config.py
lib/ansible/modules/identity/ipa/ipa_dnsrecord.py
lib/ansible/modules/identity/ipa/ipa_dnszone.py
lib/ansible/modules/identity/ipa/ipa_group.py
lib/ansible/modules/identity/ipa/ipa_hbacrule.py
lib/ansible/modules/identity/ipa/ipa_host.py
lib/ansible/modules/identity/ipa/ipa_hostgroup.py
lib/ansible/modules/identity/ipa/ipa_role.py
lib/ansible/modules/identity/ipa/ipa_service.py
lib/ansible/modules/identity/ipa/ipa_subca.py
lib/ansible/modules/identity/ipa/ipa_sudocmd.py
lib/ansible/modules/identity/ipa/ipa_sudocmdgroup.py
lib/ansible/modules/identity/ipa/ipa_sudorule.py
lib/ansible/modules/identity/ipa/ipa_user.py
lib/ansible/modules/identity/ipa/ipa_vault.py
lib/ansible/plugins/doc_fragments/ipa.py
test/sanity/ignore.txt

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @felixfontein @Andersson007

**Review by felixfontein — COMMENTED** [review]
Looks reasonable, though I have no way to verify that the types for `elements` are correct.

(This probably doesn't need a changelog fragment.)

**Comment by Akasurde** [comment]
> (This probably doesn't need a changelog fragment.)

Removed.

Will wait for @Andersson007 confirmation.

File: lib/ansible/modules/identity/ipa/ipa_dnsrecord.py [hunk 1]
```diff
@@ -40,6 +42,7 @@
     required: false
     default: 'A'
     choices: ['A', 'AAAA', 'A6', 'CNAME', 'DNAME', 'PTR', 'TXT', 'SRV', 'MX']
```
Comment (Andersson007) [inline lib/ansible/modules/identity/ipa/ipa_dnsrecord.py]:
```suggestion
    choices: ['A', 'AAAA', 'A6', 'CNAME', 'DNAME', 'PTR', 'TXT', 'SRV', 'MX']
```
i'd put them alphabetically too

File: lib/ansible/modules/identity/ipa/ipa_service.py [hunk 2]
```diff
@@ -17,28 +17,32 @@
 author: Cédric Parent (@cprh)
 short_description: Manage FreeIPA service
 description:
-- Add and delete an IPA service using IPA API
+- Add and delete an IPA service using IPA API.
 options:
   krbcanonicalname:
     description:
-    - principal of the service
+    - Principal of the service
```
Comment (Andersson007) [inline lib/ansible/modules/identity/ipa/ipa_service.py]:
```suggestion
    - Principal of the service.
```

File: lib/ansible/modules/identity/ipa/ipa_subca.py [hunk 3]
```diff
@@ -25,21 +25,24 @@
     - The Sub Certificate Authority name which needs to be managed.
     required: true
     aliases: ["name"]
+    type: str
   subca_subject:
     description:
     - The Sub Certificate Authority's Subject. e.g., 'CN=SampleSubCA1,O=testrelm.test'
```
Comment (Andersson007) [inline lib/ansible/modules/identity/ipa/ipa_subca.py]:
```suggestion
    - The Sub Certificate Authority's Subject. e.g., 'CN=SampleSubCA1,O=testrelm.test'.
```

File: lib/ansible/modules/identity/ipa/ipa_sudocmd.py [hunk 4]
```diff
@@ -24,13 +24,16 @@
     - Sudo Command.
```
Comment (Andersson007) [inline lib/ansible/modules/identity/ipa/ipa_sudocmd.py]:
```suggestion
    - Sudo command.
```

File: lib/ansible/modules/identity/ipa/ipa_user.py [hunk 5]
```diff
@@ -29,52 +30,69 @@
...
     - List of mail addresses assigned to the user.
     - If an empty list is passed all assigned email addresses will be deleted.
     - If None is passed email addresses will not be checked or changed.
+    type: list
+    elements: str
   password:
     description:
-    - Password for a user. Will not be set for an existing user unless C(update_password) is set to C(always), which is the default.
+    - Password for a user.
+    - Will not be set for an existing user unless C(update_password) is set to C(always), which is the default.
```
Comment (Andersson007) [inline lib/ansible/modules/identity/ipa/ipa_user.py]:
```suggestion
    - Will not be set for an existing user unless C(update_password) is set to C(always), which is the default.
```
good job to split bunches of text to separate lines ! Easier to read the documentation

**Review by Andersson007 — CHANGES_REQUESTED** [review]
@Akasurde thanks for tidying up the documentation!
A few cosmetic suggestions for most of the files:
1.  ```DOCUMENTATION = '''``` -> ```DOCUMENTATION = r'''``` , the same for the EXAMPLE and RETURN sections
2. Would be cool to see ```- name: description``` in the EXAMPLE section instead of using comments

File: lib/ansible/modules/identity/ipa/ipa_user.py (hunk 5 above)
Comment (tremble) [inline lib/ansible/modules/identity/ipa/ipa_user.py]:
```suggestion
    - Will not be set for an existing user unless I(update_password) is set to C(always), which is the default.
```

(or ``I(update_password=always)``)

**Review by Andersson007 — APPROVED** [review]
shipit

**Comment by Akasurde** [comment]
@felixfontein @Andersson007 Thanks for the review.

**Comment by Akasurde** [comment]
@tremble Thanks for the review.

**Pull request merged**